frontend/src/pages/groupSettings/GroupSettings.tsx

```diff
@@ -0,0 +1,244 @@
+    const fetchGroups = async () => {
+      try {
+        const groupSettings = await fetchGroupsSettings();
+        setGroupSettings(groupSettings);
+        createMissingGroupAlert(groupSettings);
+        refreshStateAfterUpdate();
+      } catch (error) {
+        if (error instanceof Error) setLoadError(error);
+      }
+    };
+
+    fetchGroups();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const handleSaveButtonClicked = async () => {
+    if (isLoadingButton) return;
+    try {
+      setIsLoadingButton(true);
+      const response = await updateGroupsSettings(groupSettings);
+      if (response.success) {
+        setGroupSettings(response.success);
+        createMissingGroupAlert(response.success);
+        dispatch(
+          addNotification({
+            status: 'success',
+            title: 'Group settings changes saved',
+            timestamp: new Date(),
+          }),
+        );
+      }
+      refreshStateAfterUpdate();
+    } catch (error) {
+      if (error instanceof Error) {
+        dispatch(
+          addNotification({
+            status: 'danger',
+            title: 'Error',
+            message: 'Error updating group settings, you might not have permission',
+            timestamp: new Date(),
+          }),
+        );
+      }
+      setIsLoadingButton(false);
+    }
+    dispatch(detectUser());
+  };
+
+  const createMissingGroupAlert = (groupsConfig: GroupsConfig) => {
+    if (groupsConfig.errorAdmin) {
+      dispatch(
+        addNotification({
+          status: 'warning',
+          title: ALERT_TITLE,
+          message: groupsConfig.errorAdmin,
+          timestamp: new Date(),
+        }),
+      );
+    }
+
+    if (groupsConfig.errorUser) {
+      dispatch(
+        addNotification({
+          status: 'warning',
+          title: ALERT_TITLE,
+          message: groupsConfig.errorUser,
+          timestamp: new Date(),
+        }),
+      );
+    }
+  };
+
+  const refreshStateAfterUpdate = () => {
+    setLoaded(true);
+    setLoadError(undefined);
+    setIsLoadingButton(false);
+    setIsGroupSettingsChanged(false);
+  };
+
+  const handleMenuItemSelection = (newState: MenuItemStatus[], field: GroupsConfigField) => {
+    switch (+field) {
+      case GroupsConfigField.ADMIN:
+        setGroupSettings({ ...groupSettings, adminGroups: newState });
+        break;
+      case GroupsConfigField.USER:
+        setGroupSettings({ ...groupSettings, allowedGroups: newState });
+        break;
+    }
+    setIsGroupSettingsChanged(true);
+  };
+
+  const isGroupEmpty = (groupList: MenuItemStatus[]): boolean => {
+    return groupList.filter((element) => element.enabled).length === 0;
+  };
+
+  return (
+    <ApplicationsPage
+      title={TITLE}
+      description={DESCRIPTION}
+      loaded={loaded}
+      empty={isEmpty}
+      loadError={loadError}
+      errorMessage={ERROR_MESSAGE}
+      emptyMessage={EMPTY_MESSAGE}
+    >
+      {!isEmpty ? (
```

Comment:
`!isEmpty && (...a component tree...)` is an easy way to avoid `boolean ? (...component tree...) : null`

React won't render `null`, `undefined`, `false` -- so lay into that. Do note it WILL render `0`... so no just `.length` checks before any ternary that doesn't have.

Reply:
i completely deleted isEmpty, yeah, never use ternary operators here.